### ISSUE TITLE
Migrate governance proposals from v1beta1 to v1

### ISSUE THREAD

**Issue by colin-axner** [cosmos/ibc-go#1282]
## Summary

Our existing governance proposals are based on the SDK version `v1beta1` of the x/gov module. Updating to `v1` of the x/gov module requires breaking changes. 

## Problem Definition

Instead of defining governance proposal handlers and proposal types, `v1` of the gov module simply executes a list of messages. This is beneficial for many reasons, but one primary one is the ability to atomically commit multiple messages in a single governance proposal. For us this could help with:
- updating many expired clients at once
- simplifying `UpgradeProposal` (currently wraps calls to x/upgrade), an IBC chain upgrade could simply be combined with a x/upgrade message
- delegation of responsibility for IBC related proposals to trusted groups (edit)

The new approach is also cleaner as there is less fragmentation in handlers/types. 

Using `v1beta1` with a [`MsgExecLegacyContent`](https://github.com/cosmos/cosmos-sdk/blob/2f6a3e6b0561ccb441d33e532665be3861c58830/x/gov/keeper/msg_server.go#L84) works alright. You can still atomically commit multiple messages with legacy code, but it does not allow for delegation of authority to trusted groups

## Proposal

Readjust our proposal handlers into msg_server handler and change our proposal types to message types. ~The fixes are primarily for code hygiene and keeping up to date with the SDK, so on the lower priority side.~ This issue should be split up into smaller issues for each proposal type. 

The existing governance handlers should be removed (and the registration on the router in simapp should be removed). We should add migration documentation explaining how to use the messages

Approved signers can be a parameter in IBC

## Issues and tasks

- [x] https://github.com/cosmos/ibc-go/issues/3672
- [x] https://github.com/cosmos/ibc-go/issues/3673
- [x] https://github.com/cosmos/ibc-go/issues/3674
- [x] https://github.com/cosmos/ibc-go/issues/3151
- [x] Update [governance proposal section](https://ibc.cosmos.network/main/ibc/proposals.html) of the docs to indicate from which version of ibc-go we support submitting proposals with the `submit-proposal` command.

____

#### For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged/assigned

**Comment by catShaark** [comment]
Can I try doing this ?

**Comment by crodriguezvega** [comment]
Hey @catShaark, thanks for asking! As Colin mentions above, this issue has a low priority for us at the moment. If you want to work on it and want to create already PRs before we merge the [PR for the upgrade to SDK 0.46](https://github.com/cosmos/ibc-go/pull/1205), then you might need to end up opening your PRs again against main once the upgrade is merged, since we definitely would want to merge these things separately to main. For that reason, it might be wise to wait a bit before taking this issue...

But we see that you are very active lately in the repo: we ❤️ it and we are very thankful for your contributions! Are you planning to sustain a certain level of commitment to work on ibc-go issues? If that would be the case then I think it would make sense if we have a chat and we can discuss what other issues you could pick up that have higher priority for us. Let me know what you think!

**Comment by catShaark** [comment]
> Hey @catShaark, thanks for asking! As Colin mentions above, this issue has a low priority for us at the moment. If you want to work on it and want to create already PRs before we merge the [PR for the upgrade to SDK 0.46](https://github.com/cosmos/ibc-go/pull/1205), then you might need to end up opening your PRs again against main once the upgrade is merged, since we definitely would want to merge these things separately to main. For that reason, it might be wise to wait a bit before taking this issue...


Ohhh, thanks for letting me know that, that's why I ask you guys so that I know if I should do it or not.


> But we see that you are very active lately in the repo: we ❤️ it and we are very thankful for your contributions! Are you planning to sustain a certain level of commitment to work on ibc-go issues? If that would be the case then I think it would make sense if we have a chat and we can discuss what other issues you could pick up that have higher priority for us. Let me know what you think!

Yess I'd love to consistently contribute to ibc-go cuz ibc is a very awesome piece of software.

**Comment by crodriguezvega** [comment]
@catShaark, are you on Discord? I would like to contact you to have a chat and Github is not the greatest platform to have a build conversation... :) I am `Carlos | Interchain#9176`. You can find me in Cosmos Network or IBC gang discord servers.

**Comment by catShaark** [comment]
> @catShaark, are you on Discord? I would like to contact you to have a chat and Github is not the greatest platform to have a build conversation... :) I am `Carlos | Interchain#9176`. You can find me in Cosmos Network or IBC gang discord servers.

I'll contact you on discord, thanks so much

**Comment by colin-axner** [comment]
When handling this issue, we should add the ability for governance to specify additional authorities. This would allow governance grant authority of specific message types. ref #3326
